### ISSUE TITLE
Can we make the wheel plugin to display 2 values?

### ISSUE THREAD

**Issue by NiFa1** [benignware/jquery-countimator#2]
Hi Rafael,

I really, really like your Countimator styled with the wheel plugin. It works very good (in an iframe) and it is by far the best progress bar I have ever seen. 

Sorry, but I have to ask you one more question ;-). Can we make it possible to get a second value in that same wheel diagram, stacked on the first value with a different color for the difference between value 1 and value 2? 

For example we have the values 80 and 90 on a scale of 100. So it should works normal till the value 80 let's say with a color red. When it reaches 80 it continues with a different color till 90. Oh it would be so powerful if we could make that happen. I really hope you are willing to give me a help in the right direction? 

Please forgive me if I asked for to much, I can ensure you that it is for a good cause where I build websites for as volunteer.

With kind regards,
Nic

**Comment by rexblack** [comment]
Hi Nic, 
thank you very much for your support and encouragement. 
I'm not sure, if i got you right. If the color of the value should change on a certain range, there's no need to display a second value. It would be sufficient to change the color at that point.
In order to achieve this, some callbacks are needed to hook into the tween.
Then it would be possible to get notified on every step and therefore toggle any css class that reflects the current state.
This is a reasonable enhancement and should have been implemented earlier. 
I'll try my best to build it into it as soon as possible.
Best regards, 
Rafael

**Comment by NiFa1** [comment]
Wow, that is a very quick and hopeful response, thanks!. I think your approach gives the result I am looking for, as I think you mean, when we use my example, that you want to display a value 90 but you want to make it possible to change the color at 80 from red to green? 

So in the example above the result will be that the wheel will be filled for 80% with red and on top of that it will be filled for 10% with green. However I hope the 80% changing value is not coded in the stylesheet, as I want to change both values (80, 90  and even the 100 at every call of the script, by using them as (post) variables in the URL.  

Thankful regards,
Nic

**Comment by NiFa1** [comment]
Hi Rafael,

Do you still remember this one and are you still willing to make that 
enhancement? If not no problem off course but it would be very nice for us.

With kind regards,
Nic

Op 27-6-2015 om 17:22 schreef Rafael Nowrotek:

**Comment by rexblack** [comment]
Hi Nic, 
i'm very sorry. Thanks for your patience. I've been into it some time ago and realized that the plugin's architecture is not adequate. So i started to rewrite things. What i came up with was something completely different and is not ready for production (You may want to take a look at dev branch if you like) 
I now promise again to include callbacks into the current version as well as samples how to achieve the desired result within the current week. 
Many thanks for your attention. 
Regards, Rafael

**Comment by NiFa1** [comment]
Hi Rafael,

No problemo off course. You don't have to do it. But if you can, you will make us really happy and thankfull!

Keep on doing your nice magic work and I hope to hear from you soon.  

With kind regards,
Nic

Verstuurd vanaf mijn telefoon.

**Comment by rexblack** [comment]
Hi Nic, 

please have a look at the current master version.
I've implemented callbacks and also ported the wheel style to svg in order to make it more flexible.
Now, css classes can be easily toggled at certain ranges in order to change color etc. on step of the animation.
Please see examples [here](http://benignware.github.io/jquery-countimator/)
Currently not supported is animating multiple values, but as i pointed out earlier, this is probably not your exact requirement. I'd prefer to keep the API consistent this time ;-)
Please inquire again, if this is still not what you needed.

Regards, 
Rafael

**Comment by NiFa1** [comment]
Hi Rafael,

Thanks for your work! However animating multiple values in the wheel was 
just where I was looking for ;-(

With kind regards,
Nic

Op 16-10-2015 om 1:34 schreef Rafael Nowrotek:

**Comment by rexblack** [comment]
Ok. I understand now. I'd really like to make this work. Technically this is not a huge thing, but I'm still in search for a consistent API.
Do you just need to display the value as digits or do you also need another arc graphics for this?

**Comment by NiFa1** [comment]
Hi Rafael,

Sorry that it took me some time to answer but I had a really bussy weekend.

You did a very nice job with the callback and changing colors and beside 
me I think a lot of people will be very happy with it. But it is not 
where I was looking for and I think I was not clear enough to tell you 
where I was looking for. I am not that good in the English language. 
Please let me try again with the picture attached.

What I have is people who create a score with doing different projects. 
Now I want to use your incredible animated Wheel to show the progress of 
all their projects just like a bar graph, but than in one wheel circle. 
See attached picture with an example of someone with 3 different projects.

I will first calculate by php-script how much projects a person has, 
what the score of each project is and what his overall goal is (or the 
total of all his projects if his goal is already exceeded. After that I 
want to use your magic Wheel-plugin to create the score circle by first 
showing the value of project 1 in color 1 (in the example green), than 
on top of that value of project 2 in color 2 (red) and after that 
project 3 in color 3 (blue) and maybe more projects after that in each 
different colors. So I want to stack all values on top of eachother and 
show them all in one wheel.

Do you think that is possible or am I asking to much?

With kind regards,
Nico

Op 16-10-2015 om 11:08 schreef Rafael Nowrotek:
